### ISSUE TITLE
[FEATURE REQUEST] Top-level H1 header is sourced from front-matter.

### ISSUE THREAD

**Issue by johnnymatthews** [ipfs/docs#430]
It'd be helpful if the top-lever headers on each page were sourced from the front-matter of the markdown document, instead of the `#` header at the start of the markdown content.

<img width="1440" alt="Screen Shot 2020-01-23 at 1 15 34 PM" src="https://user-images.githubusercontent.com/9611008/73011825-0b9e0f80-3de3-11ea-8f5e-15a43cc5f91c.png">

This feature would enable writers to format their markdown so that it fits the [Markdown Lint](https://github.com/markdownlint/markdownlint) rules, specifically [rule `MD025` _Multiple top level headers in the same document_](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md025---multiple-top-level-headers-in-the-same-document)

**Comment by cwaring** [comment]
Hold on to your 🎩because here comes a sweet hack:

```
# {{ $page.title }}
```

Use the `$page.___` to get access to any of the front matter variables inside the markdown. You're welcome 🧙‍♂️

**Comment by cwaring** [comment]
Also remember `[[ toc ]]` instead of printing a fixed ToC into the document.

**Comment by cwaring** [comment]
I should not that is sometimes useful to have these explicitly separate for reasons of SEO, so try to consider that before applying this everywhere.

**Comment by cwaring** [comment]
BTW the alternative is true so that you CAN exclude the front matter `title` key PROVIDED that there is an initial # H1`, then that key will be used as the `title` instead. It is required for one of these to be present which is why I tend to favour being verbose by having both committed to any page.

**Comment by johnnymatthews** [comment]
> `# {{ $page.title }}`

I'm not a massive fan of this hack. GitHub won't know what's going on here. It'd be better to keep things consistent across the board. If it displays on docs.ipfs.io, it should look the same way in GitHub.

> Also remember `[[ toc ]]` instead of printing a fixed ToC into the document.

Yeah, sorry that was just an artifact of editing a bunch of stuff in one go, and Markdown All In One automatically throwing a ToC in there. Silly extensions.

> you CAN exclude the front matter `title` key PROVIDED that there is an initial `# H1`

This might be the best of both worlds then! Keeps the formatting on both docs.ipfs.io and GitHub clean, and doesn't brake markdown formatting rules! Excellent!

**Comment by cwaring** [comment]
If we exclude the `title` key we have to be super diligent with content changes because if a page is later updated and the initial H1 is removed then all the SEO data will be stripped due to the missing title. Hopefully this helps you understand the small compromise of using it in both places for clarity and redundancy.

**Comment by cwaring** [comment]
Markdown currently gets passed through [Prettier](https://prettier.io/blog/2017/11/07/1.8.0.html) for auto formatting upon commit but we could also add a set of linter rules to ensure a certain style is abided by.

**Comment by johnnymatthews** [comment]
Fair enough, I don't want to break SEO all over the place. What we can do for now then it just ignore the `h1` rule from Markdown Lint.

Didn't know we were using that Prettier thing. Do you know where that config's set up? Straight off the bat I want to take the [word-wrapping](https://prettier.io/docs/en/options.html#prose-wrap) out.

**Comment by jessicaschilling** [comment]
Off-topic but does Prettier lint out strings for other strings? Cough cough "IFPS" -> "IPFS" cough.

**Comment by johnnymatthews** [comment]
I've created a `.markdownlint.json` file within the project's root. When the Markdown Lint extension is installed, the [MD025 Multiple top level headers in the same document](https://github.com/markdownlint/markdownlint/blob/master/docs/RULES.md#md025---multiple-top-level-headers-in-the-same-document) will be ignored. This file is just in the [`docs-contribution-section` branch](https://github.com/ipfs/ipfs-docs-v2/commit/981f81e0d6e446672bf0e6e3d086696ea86fd0c2) for now.

> Off-topic but does Prettier lint out strings for other strings? Cough cough "IFPS" -> "IPFS" cough.

Not sure if Prettier is the correct tools for this. For now I'll just periodically search all the docs for `IFPS`.

<img width="339" alt="Screen Shot 2020-01-27 at 1 31 39 PM" src="https://user-images.githubusercontent.com/9611008/73202604-6009fd80-4109-11ea-8512-3e2dfa181545.png">

**Comment by cwaring** [comment]
Prettier doesn't catch errors it is primarily responsible for code formatting. If want to add markdown linting then we should do it via the commit hooks otherwise it won't be caught before the CI/PR stage. Something to investigate more soon.
